src/chunk_constraint.c

```diff
@@ -444,6 +444,63 @@ typedef struct ChunkConstraintScanData
 	DimensionSlice *slice;
 } ChunkConstraintScanData;
 
+/*
+ * Scan for all chunk constraints that match the given slice ID. The chunk
+ * constraints are not saved, only the matching chunk id is returned.
+ */
+int
+ts_chunk_constraint_scan_by_dimension_slice_chunk_id(const DimensionSlice *slice, ChunkScanCtx *ctx,
+													 MemoryContext mctx)
+{
+	int result = 0;
+
+	ScanIterator iterator = ts_scan_iterator_create(CHUNK_CONSTRAINT, AccessShareLock, mctx);
```

Comment:
Consider moving this outside this function and, e.g., passing in the iterator to avoid repeated open/close.

Reply:
Done.